### PR TITLE
Add reformat rules and couple of sample files

### LINKED ISSUE #60: Tooling based formatter for consistent style and better readability
Being new to the project and Scala, I found the dense code difficult to navigate and understand. By having a tooling based formatter the styling will stay consistent with good settings for readability.

I found that scala metals vs code extension natively supports formatting by creating a `.scalafmt.conf`. The bare minimum file with default settings contains this -

```
version = "3.0.5"
```

More settings are [here](https://scalameta.org/scalafmt/docs/configuration.html).

### PR DESCRIPTION
**Pull request by twitu**
Closes #60

### CODE REVIEW

File: shared/src/main/scala/mlscript/Typer.scala [hunk 1]
```diff
@@ -73,213 +114,411 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
...
+        TypeName("unit"),
+        Nil,
+        Nil,
+        Top,
+        List.empty,
+        List.empty,
+        Set.empty,
+        N
+      ) ::
+      Nil
```
Comment (LPTK) [inline shared/src/main/scala/mlscript/Typer.scala:287]: Oh my, I'd rather not...!

File: shared/src/main/scala/mlscript/Typer.scala [hunk 2]
```diff
@@ -363,55 +619,85 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
...
+                      else
+                        f._1 ->
+                          freshVar(
+                            noProv,
+                            S(
+                              f._1.name.drop(f._1.name.indexOf('#') + 1)
+                            ) // strip any "...#" prefix
+                          )(1).tap(_.upperBounds ::= f._2)
+                    )
+                    .toList
```
Comment (LPTK) [inline shared/src/main/scala/mlscript/Typer.scala:649]: The original was fine. This just looks like obscene formatting to me 😝

**Review by LPTK — COMMENTED** [review]
Yeah, as explained in the issue I don't like it, as is. If we were a team of 20 developers I'd consider it, but right now it seems overkill and to have too big drawbacks.

Besides, now is not the time to do this as it would create lots of tricky conflicts with pending PRs, most notably https://github.com/hkust-taco/mlscript/pull/59.

File: shared/src/main/scala/mlscript/Typer.scala (hunk 1 above)
Comment (twitu) [reply shared/src/main/scala/mlscript/Typer.scala:287]: I agree this one is pretty bad.

**Comment by twitu** [comment]
Closing for further discussion on #60

**Pull request closed without merging**